### ISSUE TITLE
memory address maybe error

### ISSUE THREAD

**Issue by wangdiskater** [izarooni/MapleEzorsia#3]
when i use ida open gms083.idb i found:

:009DFE68 8D 84 06 A8 FD FF FF        lea     eax, [esi+eax-258h]   
009DFCF0 8D 84 01 E0 FC FF FF        lea     eax, [ecx+eax-320h] 

but code is:
```
Memory::WriteInt(0x009DFCF0 + 3, m_nGameHeight);//lea eax,[esi+eax-600]
Memory::WriteInt(0x009DFE68 + 3, m_nGameWidth);	//lea eax,[ecx+eax-800]  
```
i think maybe it is not right and  need exchange the address.

and 
```
Memory::WriteInt(0x008DE8EE + 2, (-m_nGameWidth + 32) / 2); //lea edi,[eax-647]           
Memory::WriteInt(0x008DE8E5 + 2, -m_nGameWidth + 228); //lea ebx,[eax-427]               
```
is error too.

idb is:
 008DE8EE 8D 98 55 FE FF FF           lea     ebx, [eax-1ABh] 
008DE8E5 8D B8 79 FD FF FF           lea     edi, [eax-287h]

**Comment by izarooni** [comment]
Thanks for creating an issue! Could you can help me understand why you think the address is incorrect?

### TL;DR first issue
I haven't looked into the details of it but, I did notice that I'm writing a positive value instead of a negative value and maybe it's causing weird behavior? Maybe try changing the value to use a negative value.
```cpp
Memory::WriteInt(0x009DFCF0 + 3, -m_nGameHeight);   //lea eax,[esi+eax-600]
Memory::WriteInt(0x009DFE68 + 3, -m_nGameWidth);    //lea eax,[ecx+eax-800]  
```
---
In the first issue we have a few things going on:
```asm
009DFCF0 ; 8D 84 01 E0 FC FF FF
lea     eax, [ecx+eax+FFFFFCE0]
```
First and foremost, the value we are looking to change is FFFFFCE0 and it's inside a function that I have named as CWnd::OnMoveWnd.

IDA can show this as an unsigned or signed value, converting this to a signed value will give us the number we're familiar with.
You can do this by right-clicking the value and selecting "change sign" or try using the default hotkey `Shift+_`
The instruction should now look like this:
```asm
lea     eax, [ecx+eax-320h]
```
The instruction has a total of 7 bytes and we know the last 4 bytes of this instruction represents the value we want to change.
This is why we are changing the value of `009DFCF0 + 3`.
```
8D : opcode (+0 offset)
84 : ecx (+1 offset)
01 : eax (+2 offset)
E0 FC FF FF : value (+3 offset)
```

I wrote a script which shows me the hexadecimal representation of a chosen value:
```javascript
const toHex = (n) => parseInt(n).toString(16)

const toAOB = (n) => {
	let array = [
		n & 0xFF,
		(n >> 8) & 0xFF,
		(n >> 16) & 0xFF,
		(n >> 24) & 0xFF
	]
	return array.map(toHex)
}


console.log('-800: ', toAOB(-800))
console.log('800: ', toAOB(800))
```
and the result is as follows, which also matches the value in the original instruction which means the value could be incorrect.
```
izarooni>node test.js
-800:  [ 'e0', 'fc', 'ff', 'ff' ]

izarooni>node test.js
800:  [ '20', '3', '0', '0' ]
```
---

### As for your second issue
```cpp
Memory::WriteInt(0x008DE8EE + 2, (-m_nGameWidth + 32) / 2); //lea edi,[eax-647]           
Memory::WriteInt(0x008DE8E5 + 2, -m_nGameWidth + 228); //lea ebx,[eax-427]   
```
I don't really know what this modifies exactly but the code seems related or similar to the quickslot window in-game.
If you could please tell me what this is affecting in-game maybe I can help more.
